### DISCUSSION TITLE
How to correctly create a mouseover effect without flickering?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1399] [Q&A] [answered]
### Question

Hi All,

I am trying to create a mouse over effect, where I will have an image. Then if you hover over the mouse some text will come up. You move away, and the text goes away. So I am using the ```on``` event and it's working fine.

The problem is the label is actually on top of the image, so imagine you hover over the image, the label comes up. Now you are moving the mouse inside the picture and come to the label. It will think as if the mouse has moved out, because now it's the label. So it's start flickering. 

I can't figure out what will be the correct solution. Like its getting to a weird circular issue. Can you please advise me on this?
Here is what I have tried so far:
```python
image = 'https://picsum.photos/id/237/200/300'

def make_visible():
    mylabel.visible = True

def make_invisible():
    mylabel.visible = False

with ui.card().tight().classes('h-40 w-40') as mycard:
    myimage = ui.image(image).classes('w-[100%] h-[100%]')
    mylabel = ui.label('A dog').classes('absolute top-10 left-14 w-[50px] h-[20px] text-center bg-white')
    mylabel.visible = False

    myimage.on('mouseenter', lambda: make_visible())
    myimage.on('mouseleave', lambda: make_invisible())
```

Really appreciate all the help. 

Thanks,
Anindya

**Comment by Anindya088** [comment]
Hi All,

I found a hack to manage this somehow, by adding a blank container of the same size and then putting the mouse movements on that container. Like this below
```python
with ui.column().classes('h-40 w-40 border border-teal-500 bg-green') as mycard:
    # myimage = ui.image(image).classes('w-[100%] h-[100%]')
    mylabel = ui.label('A dog').classes('absolute top-10 left-14 w-[50px] h-[20px] text-center bg-white')
    mylabel.visible = False
    blankcard = ui.row().classes('absolute h-40 w-40 bg-transparent border border-teal-500')

    blankcard.on('mouseenter', lambda: make_visible())
    blankcard.on('mouseleave', lambda: make_invisible())
```

But this is not a good idea, in real case, I will probably not know the real size and position of the main image. Let's say we are working on a grid, multiple rows and columns etc. 

Please let me know what would be the correct approach for this.

Thanks a lot,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
The trick is to disable pointer events for the label, e.g. using the Tailwind class `pointer-events-none`:

```py
with ui.card().tight().classes('h-40 w-40'):
    image = ui.image('https://picsum.photos/id/237/200/300').classes('w-[100%] h-[100%]')
    label = ui.label('A dog').classes('absolute top-10 left-14 w-[50px] h-[20px] text-center bg-white pointer-events-none')
    label.visible = False

    image.on('mouseenter', lambda: label.set_visibility(True))
    image.on('mouseleave', lambda: label.set_visibility(False))
```

**Reply by Anindya088** [reply]
Hi @falkoschindler , thanks so much. This is so neat! 

Aaah! I should have thought of that. I feel dumb now 😄

Thanks so much
Anindya
